### DISCUSSION TITLE
Installation issue

### DISCUSSION THREAD

**Discussion by fidovdbos** [spatie/laravel-sitemap#436] [Q&A]
When I'm trying to install using composer I get this error:

Problem 1
    - spatie/laravel-sitemap[5.9.0, ..., 5.9.1] require spatie/crawler ^5.0 -> satisfiable by spatie/crawler[5.0.0, 5.0.1, 5.0.2, v5.x-dev].
    - spatie/laravel-sitemap 5.9.2 requires spatie/crawler ^5.0|^6.0 -> satisfiable by spatie/crawler[5.0.0, 5.0.1, 5.0.2, v5.x-dev, 6.0.0, 6.0.1, 6.0.2, v6.x-dev].
    - spatie/crawler[5.0.0, ..., v5.x-dev] require guzzlehttp/psr7 ^1.4 -> found guzzlehttp/psr7[1.4.0, ..., 1.x-dev] but the package is fixed to 2.1.0 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - spatie/crawler[6.0.0, ..., v6.x-dev] require guzzlehttp/psr7 ^1.7 -> found guzzlehttp/psr7[1.7.0, ..., 1.x-dev] but the package is fixed to 2.1.0 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - Root composer.json requires spatie/laravel-sitemap ^5.9 -> satisfiable by spatie/laravel-sitemap[5.9.0, 5.9.1, 5.9.2].

I have tried to install it with --with-all-dependencies but this is not giving me a solution because other packages are using new versions of guzzle

**Comment by kirgus** [comment]
![CleanShot 2022-02-12 at 10 29 27@2x](https://user-images.githubusercontent.com/47494681/153701833-51a06fca-a6a9-45a4-9e4b-e5f459c1ab98.jpg)

Similar problem

**Comment by mro95** [comment]
Same issue here with Laravel 7.0, but 6.0 works fine

**Comment by chescos** [comment]
Same issue

**Comment by shiroze** [comment]
Until now this is still not getting answer ?

Laravel v.7.29
First I try with version 5.9 showing me this

```
Problem 1
    - spatie/laravel-sitemap[5.9.0, ..., 5.9.2] require illuminate/support ^8.0 -> found illuminate/support[v8.0.0, ..., 8.x-dev] but these were not loaded, likely because it conflicts with another require.
    - Root composer.json requires spatie/laravel-sitemap ^5.9 -> satisfiable by spatie/laravel-sitemap[5.9.0, 5.9.1, 5.9.2].
```

Then I change it to 5.7 after read the changelog that support laravel 7
```
Problem 1
    - spatie/laravel-sitemap[5.7.0, ..., 5.8.0] require spatie/crawler ^4.1.0 -> satisfiable by spatie/crawler[4.1.0, ..., v4.x-dev].
    - spatie/crawler[4.6.6, ..., v4.x-dev] require guzzlehttp/psr7 ^1.4 -> found guzzlehttp/psr7[1.4.0, ..., 1.x-dev] but the package is fixed to 2.3.0 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - spatie/crawler[4.1.0, ..., 4.6.5] require guzzlehttp/guzzle ^6.3 -> found guzzlehttp/guzzle[6.3.0, ..., 6.5.x-dev] but the package is fixed to 7.4.4 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - spatie/laravel-sitemap[5.9.0, ..., 5.9.2] require illuminate/support ^8.0 -> found illuminate/support[v8.0.0, ..., 8.x-dev] but these were not loaded, likely because it conflicts with another require.
    - Root composer.json requires spatie/laravel-sitemap ^5.7 -> satisfiable by spatie/laravel-sitemap[5.7.0, ..., 5.9.2].
```

So for me the solution that is work is
`composer require spatie/laravel-sitemap:^5.7 -W`

It's automatically downgrade and upgrade base on required dependencies.

**Comment by emrearpaci** [comment]
Delete composer.lock and try again.

**Reply by vmrfriz** [reply]
It didn't help. Laravel 10.18
